### ISSUE TITLE
What are the definitions of the output?

### ISSUE THREAD

**Issue by datavistics** [cdpierse/transformers-interpret#30]
@cdpierse , Can you give definitions of the output words? Ill put my understandings, but please correct them. 

**True Label**: The index of the label given in `class_name`?
**Predicted Label**: Output of the model
**Predicted label in parens**: I have no idea, I cant reproduce this number. Doesnt appear to be the predicted probability or the prob normalized with softmax.
**Attribution Label**: The label given in `class_name`
**Attribution Score**: Not sure. Doesnt appear to be the score of the `class_name` label.
**Legend**: How each word effects the `Attribution Label` class?

**Comment by cdpierse** [comment]
Hi @datavistics, 

Of course I'll do my best to explain. Much of what is output here is coming from the Captum package and is specifically related to Integrated Gradients and Layer Integrated Gradients. You can find out more about the algorithm [here](https://captum.ai/docs/algorithms).

**True Label**: Since I am using the visualize method from Captum this requires a "True Label" which is only available if the class is known. In many cases the True class will not be known so by default I fill in this value with the predicted class. If you know what the True label is you can pass that string value to the `visualize()` method like ` visualize(true_class = "Positive")` doing this will make True Label == `true_class`.
**Predicted Label**: Exactly this will always be the predicted class/label. 
**Predicted label numeric value in parens**: So this can be confusing to me too, but I have always taken this to be prediction probability for the label/class. This is also confirmed if you take a look at [Captums `visualize_text()`](https://github.com/pytorch/captum/blob/43acfcd320c44e8e469bf5276af9d55cf62145aa/captum/attr/_utils/visualization.py#L538) function which this package uses. This value is referred to as `pred_prob` here. 
**Attribution Label**: Yes exactly, whatever class you would like attributions for even if it is not the predicted class will be displayed in this column.
**Attribution Score**: This is the sum of the attributions for class that attributions are being calculated on. (The predicted class by default). The way I like to think about this is that the the higher the attribution score the more words there are pushing that prediction towards a class, if attribution scores are close to 0 above or below I take that as meaning that the prediction is more uncertain. 
**Legend**: Yes that's correct.

**Comment by datavistics** [comment]
Thanks for the fast reply. Most of these make sense.

> Predicted label numeric value in parens: So this can be confusing to me too, but I have always taken this to be prediction probability for the label/class. This is also confirmed if you take a look at Captums visualize_text() function which this package uses. This value is referred to as pred_prob here.

When I change the `class_name` label, I get different values for this. So, Im a little suspicious.

**Comment by cdpierse** [comment]
Nice catch. I'm setting `self.pred_probs` inside the `_forward()` method but the problem with this could be that this function gets called a bunch of times during attribution calculation as the Integrated Gradients algorithm attempts to "integrate" the baseline into the actual prediction. This could be what's leading to hard to understand `pred_probs`. My best guess is that I'll need to set `self.pred_probs` elsewhere most likely in `self.predicted_class_index` but I'll think on it. Thanks for pointing this out.
